server/block/jukebox.go

```diff
@@ -0,0 +1,128 @@
+package block
+
+import (
+	"fmt"
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/entity"
+	"github.com/df-mc/dragonfly/server/internal/nbtconv"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+	"math/rand"
+	"time"
+)
+
+// Jukebox is a block used to play music discs.
+type Jukebox struct {
+	solid
+	bass
+
+	// item is the music disc played by the jukebox.
+	item item.Stack
```

Comment:
I wonder if we should just have `Disc` exported here and only check for music discs later on. That would mean we can remove the `WithMusicDisc` and `WithoutMusicDisc` methods.